src/shell_utils.cpp

```diff
@@ -0,0 +1,169 @@
+#include "shell_utils.h"
+#include <sstream>
+#include <iostream>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+// ---- c-tor for struct Job -----
+Job::Job(const std::string& cmd, int processId, bool running)
+    : command(cmd), pid(processId), isRunning(running) {
+}
+
+std::vector<Job> backgroundJobs; // Vector to store background jobs
+
+// ----- Function definitions -----
+// Splits a string into a vector of tokens based on a delimiter
+std::vector<std::string> split(const std::string& input, char delimiter) {
+    std::vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(input);
+    while (std::getline(tokenStream, token, delimiter)) {
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+// Executes a child process with the given command path and arguments
+void executeChildProcess(const std::string& commandPath, const std::vector<std::string>& arguments) {
+    std::vector<char*> args;
+    args.push_back(const_cast<char*>(commandPath.c_str())); // First argument is the command itself
```

Comment:
Are you sure you need `const_cast` instead of making it a `std::vector<const char*>`?